### ISSUE TITLE
Clippy warning on master

### ISSUE THREAD

**Issue by tcharding** [filecoin-project/rust-fil-proofs#1113]
### Description

I'm getting a clippy warning on master, what am I doing wrong please?

```
error: using `clone` on a `Copy` type
  --> storage-proofs/core/src/gadgets/test/mod.rs:74:28
   |
74 |             to_remove.push(var.clone())
   |                            ^^^^^^^^^^^ help: try dereferencing it: `*var`
   |
```

```
$ cargo --version
cargo 1.45.0-nightly (cb06cb269 2020-05-08)
$ cargo clippy --version
clippy 0.0.212 (a74d186 2020-05-14)
```

On comit: (060bf75) "Merge pull request #1111 from filecoin-project/update-codeowners"

**Comment by tcharding** [comment]
Resolved! The cause of this was that I had an override `rustup override` set to nightly 1.45 instead of the version in this repos `rust-toolchain` (1.44).

`rustup override unset` saved me.

Excuse the noise.

**Comment by tcharding** [comment]
> Excuse the noise.

Now this is funny. I have only recently started hacking on this repo, the first thing I did was open this issue. Finally some weeks later, having worked out that the issue was with me not the repo I just closed it (we use v1.44 of cargo, I had v1.45 configured). In parallel I have been working on the removing `unwrap` [issue](https://github.com/filecoin-project/rust-fil-proofs/issues/390). I finally got rid of all the `unwrap`s this week and now went to add the new clippy lint. Now the funny bit, it was while adding the new lint that I worked out the root cause of this issue and closed it - two minutes later I find out that the new lint we need for the unwrap issue is only in v1.45 - ha!
